# Custom changes to the Herald colour scheme.
--- a/herald.vim	2010-10-18 04:00:53.000000000 +0200
+++ b/herald.vim	2025-06-06 15:23:50.784880374 +0200
@@ -21,7 +21,7 @@
 
 	" -> Text; Miscellaneous
 	hi Normal         guibg=#1F1F1F guifg=#D0D0D0 gui=none
-	hi SpecialKey     guibg=#1F1F1F guifg=#E783E9 gui=none
+	hi SpecialKey     guibg=#141414 guifg=#E783E9 gui=none
 	hi VertSplit      guibg=#1F1F1F guifg=#FFEE68 gui=none
 	hi SignColumn     guibg=#1F1F1F guifg=#BF81FA gui=none
 	hi NonText        guibg=#1F1F1F guifg=#FC6984 gui=none
@@ -51,8 +51,8 @@
 	hi ModeMsg        guibg=#1F1F1F guifg=#FFA500 gui=none
 
 	" -> Search 
-	hi Search         guibg=#FF9E16               gui=none 
-	hi IncSearch      guibg=#FF9E16               gui=none
+	hi Search         guibg=#696567 guifg=#FFEE68 gui=none
+	hi IncSearch      guibg=#696567 guifg=#FFEE68 gui=none
 
 	" -> Diff
 	hi DiffAdd        guibg=#006124 guifg=#ED9000 gui=none
@@ -78,20 +78,22 @@
 
 	" -> Code
 	hi Comment        guibg=#1F1F1F guifg=#696567 gui=none
+	hi SpecialComment guibg=#1F1F1F guifg=#696567 gui=none
+	hi Tabstop        guibg=#1F1F1F guifg=#696567 gui=none
 	hi Constant       guibg=#1F1F1F guifg=#6DF584 gui=none
 	hi String         guibg=#1F1F1F guifg=#FFB539 gui=none
 	hi Error          guibg=#1F1F1F guifg=#FC4234 gui=none
 	hi Identifier     guibg=#1F1F1F guifg=#70BDF1 gui=none
 	hi Function       guibg=#1F1F1F guifg=#90CBF1 gui=none
 	hi Ignore         guibg=#1F1F1F guifg=#1F1F1F gui=none
-	hi MatchParen     guibg=#FFA500 guifg=#1F1F1F gui=none
+	hi MatchParen     guibg=#1F1F1F guifg=#FFA500 gui=none
 	hi PreProc        guibg=#1F1F1F guifg=#BF81FA gui=none
-	hi Special        guibg=#1F1F1F guifg=#FFB539 gui=none
+	hi Special        guibg=#1F1F1F guifg=#FFEE68 gui=none
 	hi Todo           guibg=#1F1F1F guifg=#FC4234 gui=bold
 	hi Underlined     guibg=#1F1F1F guifg=#FC4234 gui=underline
 	hi Statement      guibg=#1F1F1F guifg=#E783E9 gui=none
 	hi Operator       guibg=#1F1F1F guifg=#FC6984 gui=none
-	hi Delimiter      guibg=#1F1F1F guifg=#FFEC48 gui=none
+	hi Delimiter      guibg=#1F1F1F guifg=#FC6984 gui=none
 	hi Type           guibg=#1F1F1F guifg=#FFEE68 gui=none
 	hi Exception      guibg=#1F1F1F guifg=#FC4234 gui=none
 
@@ -108,7 +110,7 @@
 
 	" -> Text; Miscellaneous
 	hi Normal         ctermbg=234  ctermfg=252   cterm=none
-	hi SpecialKey     ctermbg=234  ctermfg=176   cterm=none
+	hi SpecialKey     ctermbg=233  ctermfg=176   cterm=none
 	hi VertSplit      ctermbg=234  ctermfg=227   cterm=none
 	hi SignColumn     ctermbg=234  ctermfg=141   cterm=none
 	hi NonText        ctermbg=234  ctermfg=204   cterm=none
@@ -138,8 +140,8 @@
 	hi ModeMsg        ctermbg=234  ctermfg=214   cterm=none
 
 	" -> Search 
-	hi Search         ctermbg=214                cterm=none 
-	hi IncSearch      ctermbg=214                cterm=none
+	hi Search         ctermbg=241  ctermfg=227   cterm=none
+	hi IncSearch      ctermbg=241  ctermfg=227   cterm=none
 
 	" -> Diff
 	hi DiffAdd        ctermbg=22   ctermfg=208   cterm=none
@@ -165,20 +167,22 @@
 
 	" -> Code
 	hi Comment        ctermbg=234  ctermfg=241   cterm=none
+	hi SpecialComment ctermbg=234  ctermfg=241   cterm=none
+	hi Tabstop        ctermbg=234  ctermfg=241   cterm=none
 	hi Constant       ctermbg=234  ctermfg=84    cterm=none
 	hi String         ctermbg=234  ctermfg=215   cterm=none
 	hi Error          ctermbg=234  ctermfg=203   cterm=none
 	hi Identifier     ctermbg=234  ctermfg=75    cterm=none
 	hi Function       ctermbg=234  ctermfg=117   cterm=none
 	hi Ignore         ctermbg=234  ctermfg=234   cterm=none
-	hi MatchParen     ctermbg=214  ctermfg=234   cterm=none
+	hi MatchParen     ctermbg=234  ctermfg=214   cterm=underline
 	hi PreProc        ctermbg=234  ctermfg=141   cterm=none
-	hi Special        ctermbg=234  ctermfg=215   cterm=none
+	hi Special        ctermbg=234  ctermfg=227   cterm=none
 	hi Todo           ctermbg=234  ctermfg=203   cterm=bold
 	hi Underlined     ctermbg=234  ctermfg=203   cterm=underline
 	hi Statement      ctermbg=234  ctermfg=176   cterm=none
 	hi Operator       ctermbg=234  ctermfg=204   cterm=none
-	hi Delimiter      ctermbg=234  ctermfg=227   cterm=none
+	hi Delimiter      ctermbg=234  ctermfg=204   cterm=none
 	hi Type           ctermbg=234  ctermfg=227   cterm=none
 	hi Exception      ctermbg=234  ctermfg=203   cterm=none 
 
@@ -195,7 +199,7 @@
 
 	" -> Text; Miscellaneous
 	hi Normal         ctermbg=8  ctermfg=15 cterm=none
-	hi SpecialKey     ctermbg=8  ctermfg=5  cterm=none
+	hi SpecialKey     ctermbg=7  ctermfg=5  cterm=none
 	hi VertSplit      ctermbg=8  ctermfg=14  cterm=none
 	hi SignColumn     ctermbg=8  ctermfg=5  cterm=none
   hi NonText        ctermbg=8  ctermfg=4  cterm=none
@@ -225,8 +229,8 @@
   hi ModeMsg        ctermbg=8  ctermfg=14 cterm=none
 
   " -> Search 
-  hi Search         ctermbg=6             cterm=none 
-  hi IncSearch      ctermbg=6             cterm=none
+  hi Search         ctermbg=7  ctermfg=14 cterm=none
+  hi IncSearch      ctermbg=7  ctermfg=14 cterm=none
 
   " -> Diff
   hi DiffAdd        ctermbg=0  ctermfg=10 cterm=none
@@ -252,20 +256,22 @@
 
   " -> Code
   hi Comment        ctermbg=8  ctermfg=7  cterm=none
+  hi SpecialComment ctermbg=8  ctermfg=7  cterm=none
+  hi Tabstop        ctermbg=8  ctermfg=7  cterm=none
   hi Constant       ctermbg=8  ctermfg=10 cterm=none
   hi String         ctermbg=8  ctermfg=6  cterm=none
   hi Error          ctermbg=8  ctermfg=4  cterm=none
   hi Identifier     ctermbg=8  ctermfg=11 cterm=none
   hi Function       ctermbg=8  ctermfg=11 cterm=none
   hi Ignore         ctermbg=8  ctermfg=8  cterm=none
-  hi MatchParen     ctermbg=14 ctermfg=8  cterm=none
+  hi MatchParen     ctermbg=8  ctermfg=14 cterm=underline
   hi PreProc        ctermbg=8  ctermfg=5  cterm=none
-  hi Special        ctermbg=8  ctermfg=12 cterm=none
+  hi Special        ctermbg=8  ctermfg=14 cterm=none
   hi Todo           ctermbg=8  ctermfg=12 cterm=bold
   hi Underlined     ctermbg=8  ctermfg=12 cterm=underline
   hi Statement      ctermbg=8  ctermfg=13 cterm=none
   hi Operator       ctermbg=8  ctermfg=4  cterm=none
-  hi Delimiter      ctermbg=8  ctermfg=14 cterm=none
+  hi Delimiter      ctermbg=8  ctermfg=4  cterm=none
   hi Type           ctermbg=8  ctermfg=14 cterm=none
   hi Exception      ctermbg=8  ctermfg=12 cterm=none
 
@@ -283,7 +289,7 @@
 
 	" -> Text; Miscellaneous
 	hi Normal         ctermbg=8  ctermfg=7  cterm=none
-	hi SpecialKey     ctermbg=8  ctermfg=5  cterm=none
+	hi SpecialKey     ctermbg=7  ctermfg=5  cterm=none
 	hi VertSplit      ctermbg=8  ctermfg=6  cterm=none
 	hi SignColumn     ctermbg=8  ctermfg=5  cterm=none
   hi NonText        ctermbg=8  ctermfg=4  cterm=none
@@ -313,8 +319,8 @@
   hi ModeMsg        ctermbg=8  ctermfg=6  cterm=none
 
   " -> Search 
-  hi Search         ctermbg=6             cterm=none 
-  hi IncSearch      ctermbg=6             cterm=none
+  hi Search         ctermbg=7  ctermfg=6  cterm=none
+  hi IncSearch      ctermbg=7  ctermfg=6  cterm=none
 
   " -> Diff
   hi DiffAdd        ctermbg=0  ctermfg=2  cterm=none
@@ -340,20 +346,24 @@
 
   " -> Code
   hi Comment        ctermbg=8  ctermfg=7  cterm=none
+  hi SpecialComment ctermbg=8  ctermfg=7  cterm=none
+  hi Tabstop        ctermbg=8  ctermfg=7  cterm=none
   hi Constant       ctermbg=8  ctermfg=2 cterm=none
   hi String         ctermbg=8  ctermfg=6  cterm=none
   hi Error          ctermbg=8  ctermfg=4  cterm=none
   hi Identifier     ctermbg=8  ctermfg=3 cterm=none
   hi Function       ctermbg=8  ctermfg=3 cterm=none
   hi Ignore         ctermbg=8  ctermfg=8  cterm=none
-  hi MatchParen     ctermbg=6  ctermfg=8  cterm=none
+  hi MatchParen     ctermbg=8  ctermfg=6  cterm=underline
   hi PreProc        ctermbg=8  ctermfg=5  cterm=none
   hi Special        ctermbg=8  ctermfg=4 cterm=none
+  hi Special        ctermbg=8  ctermfg=6 cterm=none
   hi Todo           ctermbg=8  ctermfg=4 cterm=bold
   hi Underlined     ctermbg=8  ctermfg=4 cterm=underline
   hi Statement      ctermbg=8  ctermfg=5 cterm=none
   hi Operator       ctermbg=8  ctermfg=4  cterm=none
   hi Delimiter      ctermbg=8  ctermfg=6 cterm=none
+  hi Delimiter      ctermbg=8  ctermfg=4 cterm=none
   hi Type           ctermbg=8  ctermfg=6 cterm=none
   hi Exception      ctermbg=8  ctermfg=4 cterm=none
 
@@ -378,8 +388,15 @@
 
 " Spellcheck formatting
 if has("spell")
-	hi SpellBad   guisp=#FC4234 gui=undercurl
-	hi SpellCap   guisp=#70BDF1 gui=undercurl
-	hi SpellLocal guisp=#FFEE68 gui=undercurl
-	hi SpellRare  guisp=#6DF584 gui=undercurl
+	if &t_Co == 8
+		hi SpellBad   term=reverse  ctermbg=1
+		hi SpellCap   term=standout ctermfg=7 ctermbg=4
+		hi SpellLocal ctermfg=0     ctermbg=6
+		hi SpellRare  ctermfg=0     ctermbg=5
+	else
+		hi SpellBad   guisp=#FC4234 gui=undercurl
+		hi SpellCap   guisp=#70BDF1 gui=undercurl
+		hi SpellLocal guisp=#FFEE68 gui=undercurl
+		hi SpellRare  guisp=#6DF584 gui=undercurl
+	endif
 endif
